src/components/Editor/SearchBar.js

```diff
@@ -350,15 +364,15 @@ const SearchBar = React.createClass({
     } = this.props;
 
     if (query === "") {
-      this.setState({ enabled: true });
+      this.props.toggleFileSearch(true);
     }
 
     if (index == -1) {
-      clearIndex(ctx, query, modifiers);
+      clearIndex(ctx, query, modifiers.toJS());
```

Comment:
These `toJS` calls are temporary, once more state is lifted the utils could be refactored to access the immutable types via their APIs.